cunumeric/_ufunc/ufunc.py

```diff
@@ -203,6 +264,14 @@ def __call__(
     ):
         x = convert_to_cunumeric_ndarray(x)
 
+        if len(args) > 0:
+            if out is not None:
+                raise TypeError(
+                    "cannot specify 'out' as both a positional and keyword "
+                    "argument"
+                )
+            out = args
```

Comment:
Shouldn't this be something like `out = args[0]`? Possibly accompanied by a check that `len(args) < 2`?

Note that a cleaner way to achieve a similar affect might be to use `__call__(self, x, *, out=None, ...)` to require that `out` and all subsequent arguments be provided using keyword arguments. Or (I think) `__call__(self, x, out=None, *, ...)` to be a little more lenient and allow `out` to also be positional.

Reply:
> Shouldn't this be something like out = args[0]? Possibly accompanied by a check that len(args) < 2?

That was to reuse the [argument checking 
https://github.com/nv-legate/cunumeric/pull/315/files/ab6b1fc5bfd343a0c1281e43f1a8a238ef1f77f4#diff-dc4602a545cbb84b22aff9dcc1447e6763980e9d69af1abe1d7d9966d099c470R277

Knowing that this ufunc only takes one output, we could potentially simplify

Reply:
> Shouldn't this be something like out = args[0]? Possibly accompanied by a check that len(args) < 2?

That was to reuse the [argument checking](https://github.com/nv-legate/cunumeric/pull/315/files/ab6b1fc5bfd343a0c1281e43f1a8a238ef1f77f4#diff-dc4602a545cbb84b22aff9dcc1447e6763980e9d69af1abe1d7d9966d099c470R277) in the subsequent if statement.

> Or (I think) __call__(self, x, out=None, *, ...) to be a little more lenient and allow out to also be positional.

Knowing `cunumeric.unary_ufunc` only having one output, we could potentially do that, but I'm hoping that one day we could refactor all the ufuncs into one template, so I'd rather keep them having a similar structure, unless you object. Note that with the proposed change, the user will see a different error message when `out` is provided as both a positional argument and a keyword argument.

Reply:
Actually please ignore my comments on this, your code is consistent with how NumPy works:

```
x = np.ones(4)
z = np.zeros(4)
w = np.zeros(4)
np.frexp(x, out=(z,w)) # works
np.frexp(x, z, w) # works
# np.frexp(x, (z, w)) # fails
```

Reply:
Actually, this is working for a different reason in NumPy: the signature for `frexp` allows `out1` and `out2` to be positional, but not `out`:

```
frexp(x[, out1, out2], / [[, out, where, ...])
```

(from https://numpy.org/doc/stable/reference/ufuncs.html)

This documentation also seems to mirror my suggestion to only allow `out` to be a keyword argument.

Reply:
What exactly is your suggestion again?

Reply:
I suggest that you only allow


Essentially you will be emulating this signature:

```
```

 
If 




I think with the current way of doing things you can't pass `where` as a positional parameter, because `out` will always consume all remaining arguments

```
cn.frexp(x, a, b, c)

proper way is to support out1= and out2=

We should either 

or make it explicit using `*` in the argument list that `where` and subsequent 

we can deal with this on another issue

Reply:
I suggest we only allow up to however many outputs each operation accepts, mirroring NumPy:

```
x = np.ones(4)
z = np.zeros(4)
w = np.zeros(4)
np.frexp(x, out=(z,w), where=False) # works
np.frexp(x, z, w, where=True) # works
np.frexp(x, z, w, True) # fails with TypeError: frexp() takes from 1 to 3 positional arguments but 4 were given
```

e.g. doing something like this:

```
       if len(args) > 0:
             if out is not None:
                 raise TypeError(
                     "cannot specify 'out' as both a positional and keyword "
                     "argument"
                 )
             if len(args) > self.nout:
                 raise TypeError(
                     "too many output arrays"
                 )
             out = args
```

Reply:
Alright. I got tired of maintaining three copies of operand handling that are only slightly different, so I unified and generalized them. The logic should now be general enough to handle any ufuncs and also address #342. Please take a look.